src/librustc/ty/relate.rs

```diff
@@ -700,6 +700,16 @@ impl<'tcx> Relate<'tcx> for Kind<'tcx> {
             (UnpackedKind::Lifetime(unpacked), x) => {
                 bug!("impossible case reached: can't relate: {:?} with {:?}", unpacked, x)
             }
+            (UnpackedKind::Const(ty::Const { ty, .. }), UnpackedKind::Type(b_ty)) => {
```

Comment:
You shouldn't need all these cases. You should just need to cover `Const` with `Const` and make `Const` with anything else `bug!()`, i.e. copy the implementation here: https://github.com/rust-lang/rust/pull/59008/files#diff-af1c8bc6ae3623b749b5aebeaa1e0bffR777.

Reply:
(Note that #59008 is probably still a way from being merged, so it's worth making this change now regardless.)